projects/step-frontend/src/lib/modules/timeseries/chart/time-series-chart.component.ts

```diff
@@ -46,7 +55,14 @@ export class TimeSeriesChartComponent implements OnInit, AfterViewInit, OnChange
     }
   }
 
-  ngAfterViewInit(): void {
+  resetZoom(): void {
+    this.uplot.setData(this.uplot.data, true);
+  }
+
+  /**
+   * @param settings
+   */
+  createChart(settings: TSChartSettings): void {
     let getSize = () => {
```

Comment:
We will have to check the performance in production cases.
Right now chart rendering works inside of angular's change detection cycle.
Probably it make sense to move it's invocation in zone.runOutsideAngular() to split charts rendering from angular's underhood stuff.

Reply:
this was initially my plan. @neogucky told me to do it without, and think about it just if we have performance issues. currently it works good. I will create a task for it so we don't forget